Sources/HTTPLayer/Service/RequestService.swift

```diff
@@ -32,18 +32,21 @@ private struct SessionContainer {
 
 final class RequestService: NSObject {
 
-    fileprivate let fileManager: FileManagerProtocol
+    private let fileManager: FileManagerProtocol
+
+    private let sessionQueue = DispatchQueue.global(qos: .background)
 
     //MARK: - Handling multiple tasks
     private var currentTasks = [URLSessionTask: (request: HttpRequest, response: HttpResponse?)]()
 
     ///Sets request for current task
-    fileprivate func setCurrent(_ request: HttpRequest, for task: URLSessionTask) {
+    private func set(_ request: HttpRequest, for task: URLSessionTask) {
```

Comment:
You can consider using `subscript` instead of function.

Reply:
Problem to solve here is a little bit more complex. I plan to refactore task managing but I don't thing that  subscript will be enough.